frontend/src/metabase/query_builder/actions/downloading.ts

```diff
@@ -113,41 +124,53 @@ const getDatasetParams = ({
     return {
       method: "POST",
       url: `/api/card/${cardId}/query/${type}`,
-      params: new URLSearchParams({
-        parameters: JSON.stringify(result?.json_query?.parameters ?? []),
-      }),
+      params: new URLSearchParams({ format_rows }),
+      body: {
+        parameters: result?.json_query?.parameters ?? [],
+      },
     };
   }
 
   return {
-    url: `/api/dataset/${type}`,
     method: "POST",
-    params: new URLSearchParams({
-      query: JSON.stringify(_.omit(result?.json_query ?? {}, "constraints")),
-      visualization_settings: JSON.stringify(visualizationSettings ?? {}),
-    }),
+    url: `/api/dataset/${type}`,
+    params: new URLSearchParams({ format_rows }),
+    body: {
+      query: _.omit(result?.json_query ?? {}, "constraints"),
+      visualization_settings: visualizationSettings ?? {},
+    },
   };
 };
 
-export function getDatasetDownloadUrl(url: string) {
-  // make url relative if it's not
-  url = url.replace(api.basename, "");
-  const requestUrl = new URL(api.basename + url, location.origin);
-
+export function getDatasetDownloadUrl(url: string, params?: URLSearchParams) {
+  url = url.replace(api.basename, ""); // make url relative if it's not
+  url = api.basename + url;
+  if (params) {
+    url += `?${params.toString()}`;
+  }
+  const requestUrl = new URL(url, location.origin);
   return requestUrl.href;
 }
 
 const getDatasetResponse = ({
   url,
   method,
+  body,
   params,
 }: DownloadQueryResultsParams) => {
-  const requestUrl = getDatasetDownloadUrl(url);
+  const requestUrl = getDatasetDownloadUrl(url, params);
 
   if (method === "POST") {
-    return fetch(requestUrl, { method, body: params });
+    // BE expects the body to be form-encoded :(
+    const formattedBody = new URLSearchParams();
+    if (body != null) {
+      for (const key in body) {
+        formattedBody.append(key, JSON.stringify(body[key]));
+      }
+    }
+    return fetch(requestUrl, { method, body: formattedBody });
   } else {
-    return fetch(`${requestUrl}?${params}`);
+    return fetch(requestUrl);
```

Comment:
Had to refactor this function in [`519d44a`](https://github.com/metabase/metabase/pull/40643/commits/519d44a49ba3a92002f3eaa3dc2e3bb2eec59a09). Before it was only accepting `params` and it'd use them as a body for POST requests and as URL query params for GET requests. Now we always want a `format_rows` URL query parameter, so I changed it to accept `body` and `params` separetely